tools/docker/builder/openwrt/Dockerfile

```diff
@@ -46,68 +62,40 @@ RUN git clone "$OPENWRT_REPOSITORY" openwrt \
 
 WORKDIR /home/openwrt/openwrt
 
-COPY --chown=openwrt:openwrt configs/sdk_${SDK_CONFIG}_config /home/openwrt/sdk_config
+COPY --chown=openwrt:openwrt configs/base_${BASE_CONFIG}_config /home/openwrt/base_config
 
-RUN if [ -n "$INTEL_FEED" ] ; then \
+# We first install from the intel feed preferably, because
+# it replaces some packages from the "packages" feed.
+# If no intel feed is present, nothing will be done so it's still ok.
+RUN echo "src-git prpl ${PRPL_FEED}" >> feeds.conf.default \
+    && if [ -n "$INTEL_FEED" ] ; then \
         echo "src-git intel $INTEL_FEED" >> feeds.conf.default \
         && scripts/feeds update intel \
-        && scripts/feeds install intel_mips ; \
+        && scripts/feeds install intel_mips \
+        && scripts/feeds install -a -p intel ; \
     fi ;\
-    cp /home/openwrt/sdk_config .config \
-    && echo "CONFIG_SDK=y" >> .config \
-    && echo "CONFIG_TARGET_${TARGET}=y" >> .config \
+    scripts/feeds update -a \
+    && scripts/feeds install -a \
+    && cp /home/openwrt/base_config .config \
+    && echo "CONFIG_TARGET_${TARGET_SYSTEM}=y" >> .config \
+    && echo "CONFIG_TARGET_${TARGET_SYSTEM}_${SUBTARGET}=y" >> .config \
+    && echo "CONFIG_TARGET_${TARGET_SYSTEM}_${SUBTARGET}_${TARGET_PROFILE}=y" >> .config \
+    && echo "CONFIG_PACKAGE_prplmesh${PRPLMESH_VARIANT}=y" >> .config \
```

Comment:
Are you refering to this PR? https://github.com/prplfoundation/prplMesh/pull/917
Yes I was going to suggest not to add another Dockerfile.
If the goal is to be able to compile a development version of prplWrt (by that I mean with changes not yet part of the repository), we could mount a directory instead of cloning the repository.
It's not strictly necessary to move this logic to a build script to do that, but it may be more convenient to debug build failures.

Reply:
Yes, #917 is good example.
Could you please say what is purpose to build separate docker image for each target/subtarget/... for each commit?

How do you think will be better to have a single docker image with build environment (gcc, make, etc.) required to build OpenWrt and keep workspaces with prplWrt/OpenWrt sources on host?

Reply:
Each docker image contains a pre-built OpenWrt tree for a specific target device that we use to produce prplMesh ipks.
The ipks are then deployed to the different devices that we have before the CI tests are run.

Note that the docker images are only built once on each CI runner we have, then the build result is cached and re-used for later runs, so we don't rebuild the full docker image every time.
The images are only rebuilt from a previous step if something changes, like the OpenWrt version hash, the base configuration, and so on.

The single docker image you refer to already exist, it's just not kept in CI.
If you look at the `openwrt-prerequisites` stage it's exactly that: it contains all you need to build with OpenWrt (you can build it using the `--target` option of `docker build`).

If you need to work on OpenWrt and want to compile a development version locally, you could adapt the existing scripts and the Dockerfile to use a local OpenWrt tree instead of a version hash.
In that case, yes, you might want to move the lines we are commenting on here to a separate script for convenience, in order to build prplWrt when running the container and not when building it.

However: if you only want to build prplMesh and/or a full image for an OpenWrt target based on a public version of OpenWrt, there is no need to do all that.
Just running `tools/docker/builder/openwrt/build.sh -d <target>` will produce both the full image (when this is merged into master) and the prplMesh ipk.
It will take some time to complete on the first run because it has to build everything from scratch, then it will be much quicker on the next runs because it will only rebuild prplMesh.

Reply:
I understood described use case.

So if presence of big images with built OpenWrt in docker cache is suitable then i'd like to withdraw my offer.